packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -137,6 +137,7 @@ function GridEditSingleSelectCell(props: GridRenderEditCellParams & Omit<SelectP
       fullWidth
       {...other}
       {...rootProps.componentsProps?.baseSelect}
+      native={false}
```

Comment:
> Another solution is to support native select by adding an argument isNative to renderSingleSelectOptions such that it can switch between <MenuItem/> and <option />

+1 for this option. It will be odd to have all selects working with `native=false` and only this one that doesn't.

Reply:
I chose the latter solution(support native select).
I tested the change using Storybook, and I found that `NativeSelect` casts the value to a string.
To fix that issue, I copied the util from `filterPanelUtils` and formatted the value.
```ts
  const handleChange = async (event: React.KeyboardEvent<HTMLInputElement>) => {
    setOpen(false);
    const target = event.target as HTMLInputElement;
    // NativeSelect casts the value to a string.
    const formattedTargetValue = getValueFromValueOptions(target.value, valueOptionsFormatted);
    const isValid = await api.setEditCellValue({ id, field, value: formattedTargetValue }, event);
    ...
```